src/gl/texture.cpp

```diff
@@ -94,7 +95,7 @@ void gl_bind_texture_set(struct texture_set *_texture_set)
 {
 	VOBJ(texture_set, texture_set, _texture_set);
 
-	if(VPTR(texture_set))
+	if(VPTR(texture_set) && VREF(texture_set, texturehandle))
```

Comment:
What was the reason for this change? Be aware that this code must work on both FF7 and FF8 which not always share the same structure, although being very similar.

Reply:
I added this to check that the texture handle is valid before accessing the array.
The reason being that when rendering the deferred models as implemented in deferred.cpp I was getting an access violation here. You can test it if you remove this check, load the game, access the menu, then close.
I suppose that this is caused because when saving the render state with gl_save_state it does not make a copy of the contents of ff7_texture_set::texturehandle and under certain conditions the memory is deleted before use.

There may be a better way to prevent this problem so please let me know if you have some idea.

Reply:
Interesting finding, not sure why no one was able to trigger this before. I'll try to reproduce and let you know, thanks!

Reply:
I have investigated more about this problem and was able to fix it without modifying texture.cpp.
As said before, when closing the menu there seems to be some texture unloading and the gl_check_deferred function in deferred.cpp is called. This function deletes the memory allocated and invalidates the draw calls. I just added a check in the deferred draw loop to omit invalidated draw calls.
Before the changes of this PR only a few special case draw calls needed to be deferred so in those cases the problem described above simply does not occur.

Reply:
Glad you found your way out of this. Sorry but I'm a bit short of time in this period. Thank you, really appreciate it!